### PR TITLE
Feat: Add subdivisions for all countries

### LINKED ISSUE #2821: Crypto-to-fiat: Add country subdivisions for all countries
## Specification

This issue is to modify the state dropdown in bank details modal to show country subdivision options for all countries that have them (and not just US):
image

When creating/updating USD bank accounts, the API requires a `state` property which needs to be an ISO 3166-2 subdivision code ([Wiki](https://en.wikipedia.org/wiki/ISO_3166-2)). 

To get the data, a library such as https://github.com/dustyhoppe/iso-3166-2 can be used. File for each country should be lazily loaded when it is selected.

**Filtering UK options**
The UK has a long list of different level subdivisions (221 items), in order to provide a more streamlined experience the options should be narrowed down to the following codes:
- `GB-ENG` England
- `GB-NIR` Northern Ireland
- `GB-SCT` Scotland
- `GB-WLS` Wales

### This issue is complete when...

- [ ] Address form dropdown displays options for the selected country (if available)
- [ ] Correct `state` value is sent in mutation payload
- [ ] Existing hardcoded list `src/constants/us-states.ts` is removed

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

Don't be put off by the large file count, 195 of the 200 are .json files for the subdivision data.

This PR adds subdivision options for the countries that have them to the state selector on the bank details modal.

The form validation has been tweaked to make state a required field when the selected country has state data, and not required for when the selected country does not.

## Testing

Step 1 - Navigate to the Crypto to fiat tab of the account section: [http://localhost:9091/account/crypto-to-fiat](http://localhost:9091/account/crypto-to-fiat)

Step 2 - Complete the KYC Verification, or adjust `getCTAScheme` in `src/components/frame/v5/pages/UserCryptoToFiatPage/partials/BankDetails/consts.ts` to immediately return this without checking for bankAccountData:

```
    return {
      ctaTitle: {
        id: `${displayName}.body.ctaTitle`,
        defaultMessage: 'Update details',
      },
    };
  ```

Step 3 - Click 'Add details' on the 'Bank details' row.

Step 4 - Complete the first stage of the form.

Step 5 - Select a country and try submitting the form. If the country has subdivision data, then we should expect to see the "state is required" error validation:

 
Screenshot 2024-08-07 at 14 47 50

Step 6 - Select a country which does not have subdivision data (such as American Samoa), we should expect not to see the error validation:
 
Screenshot 2024-08-07 at 14 47 58

Step 7 - Select 'United Kingdom', we should expect to only see 4 options (the available options for United Kingdom has been restricted as per the requirements in the issue: #2821)

Screenshot 2024-08-07 at 12 44 03

## Diffs

**New stuff** ✨

* New `getSubdivisionsByCountryCode` utility function
* Subdivision json files

**Changes** 🏗

* `SubdivisionSelect` now uses `getSubdivisionsByCountryCode` utility function
* Contact details form validation state field is now required / not required dependant on selected country

**Deletions** ⚰️

* `us-states.ts` is no more

Resolves #2821

### CODE REVIEW

**Review by jakubcolony — DISMISSED** [review]
Nicely done @iamsamgibbs, all works as expected. Thanks for picking up that issue ⭐ 

https://github.com/user-attachments/assets/2f68e8d5-4ee9-4617-a663-4689500d1ce5

Probably the best proof it works is the fact Bridge validates the address as correct when creating/updating bank account:

https://github.com/user-attachments/assets/44679a21-21e8-4730-b518-33caf3dcb416

**Review by mmioana — CHANGES_REQUESTED** [review]
Awesome work on this one @iamsamgibbs 🥇 

Tested and can confirm the State options update properly on country change
Screenshot 2024-08-08 at 08 57 40
Screenshot 2024-08-08 at 08 57 47
Screenshot 2024-08-08 at 08 58 28
Screenshot 2024-08-08 at 08 58 34
Screenshot 2024-08-08 at 09 02 29
Screenshot 2024-08-08 at 09 02 33

A tiny thing I would add here is to re-trigger validation upon country change as you can select a country that has states -> hit submit -> state is a required field -> change to a country not having states -> error message is still shown until you hit submit again
I did something similar for the postcode [here](https://github.com/JoinColony/colonyCDapp/pull/2798/files#diff-b6a57a5316ebddde53e408f289559c46b8253fa1439b0613385d43f097f28673L30-L33)

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx [hunk 1]
```diff
@@ -25,11 +25,21 @@ export const CountrySelect = () => {
...
 
   const handleSelect = () => {
     // if country changed user should choose state of new country
     setValue('state', '');
+
+    // If no attempt to submit the form has been made yet, do not trigger state validation
+    if (isSubmitted) {
+      return;
+    }
+    trigger('state');
```
[the remark this answers is not in the document]
Comment (mmioana) [reply src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx:42]: Awesome @iamsamgibbs! Surely the `isSubmitted` check will be helpful, though from my testing if the form wasn't submitted, it didn't display the error message for the postcode 🤔 
Either way once your PR is in, the magic will be in place also for the postcode field 🎉

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx [hunk 2]
```diff
@@ -25,11 +25,21 @@ export const CountrySelect = () => {
...
+    trigger,
+    formState: { isSubmitted },
+  } = useFormContext();
 
   const handleSelect = () => {
     // if country changed user should choose state of new country
     setValue('state', '');
+
+    // If no attempt to submit the form has been made yet, do not trigger state validation
+    if (isSubmitted) {
```
Comment (mmioana) [inline src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx:39]: I think here the condition should be ` if (!isSubmitted) {` otherwise I ran into this situation
No country is selected
Screenshot 2024-08-08 at 10 53 16
Select country -> State validation is already triggered
Screenshot 2024-08-08 at 10 53 21
Error is still shown after selecting a value for the state
Screenshot 2024-08-08 at 10 53 29

**Review by mmioana — CHANGES_REQUESTED** [review]

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx (hunk 2 above)
Comment (iamsamgibbs) [reply src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/CountrySelect.tsx:39]: 🤦‍♂️ Yes of course

**Review by mmioana — APPROVED** [review]
Works like a charm @iamsamgibbs 🚀 

Before submitting
Screenshot 2024-08-08 at 11 37 01
Screenshot 2024-08-08 at 11 39 23
Screenshot 2024-08-08 at 11 39 29
After triggering form 'Submit'
Screenshot 2024-08-08 at 11 37 10
Screenshot 2024-08-08 at 11 37 22
Screenshot 2024-08-08 at 11 37 30
Screenshot 2024-08-08 at 11 37 37
Screenshot 2024-08-08 at 11 37 45

**Review by jakubcolony — APPROVED** [review]

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/validation.ts [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
 export const addressValidationSchema = Yup.object({
+  country: Yup.string().required(),
   address1: Yup.string().required(),
   address2: Yup.string().notRequired(),
   city: Yup.string().required(),
-  state: Yup.string().required(),
+  state: Yup.string().when('country', {
+    is: (country) => COUNTRIES_WITHOUT_STATES.includes(country),
+    then: Yup.string().notRequired(),
+    otherwise: Yup.string().required(),
```
Comment (rumzledz) [inline src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/validation.ts:13]: This is more of a product question and better to do in a different PR but I wonder if we should also remove the State field if there are no options for it to begin with 🤔

**Review by rumzledz — APPROVED** [review]
Great work on this one @iamsamgibbs! ✨ 

The default field validations are in tact:
Screenshot 2024-08-08 at 12 14 59

American Samoa is indeed not requiring the state field:
Screenshot 2024-08-08 at 12 15 08

United Kingdom is only showing 4 selectable items in the State field:
Screenshot 2024-08-08 at 12 15 31

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/validation.ts (hunk 3 above)
Comment (iamsamgibbs) [reply src/components/frame/v5/pages/UserCryptoToFiatPage/partials/ContactDetailsForm/validation.ts:13]: I agree that is a good idea for a future PR 👍
